### PR TITLE
Clarify docs about having to resolve incoming conflicts

### PR DESCRIPTION
**Pull request by leplatrem**
While working on the [kinto boiler plate app](https://github.com/Kinto/kinto-react-boilerplate), I realized that I misunderstood the meaning of ``CLIENT_WINS``. 

This PR is an attempt to clarify that **incoming** conflicts still may occur with the `CLIENT_WINS` policy.

@Natim @ametaireau @n1k0 r?

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -217,39 +217,51 @@ articles.sync()
...
+import Collection from "kinto/lib/collection";
+
 articles.sync({
   strategy: Collection.strategy.CLIENT_WINS,
   headers: {Authorization: "Basic bWF0Og=="}
 })
-  .then(console.log.bind(console));
+  .then(result => {
+    console.log(result);
+  })
```
Comment (n1k0) [inline docs/api.md]: Hmm, why this change?

**Comment by n1k0** [comment]
Looks good though I feel like we should re-implement default behavior for automatic incoming conflict handling according to the global strategy mode used — like it was done some times ago.

That way, users selecting a given strategy would get consistent behavior all along the whole sync flow, which would make their experience more simple & more consistent. Thoughts?

File: docs/api.md (hunk 1 above)
Comment (leplatrem) [inline docs/api.md]: Just to explicit that the *result* we're talking about is coming here.
I had the feeling that newcomers might not get always get along with JavaScript brevity. But I might be wrong, and would be willing to revert this change :)

**Comment by leplatrem** [comment]
This is exactly what was discussed in the [boilerplate PR](https://github.com/Kinto/kinto-react-boilerplate/pull/13).

Instead of changing the documentation, we might prefer to apply the policy for incoming conflicts as you said.

* ``CLIENT_WINS``: no precondition headers in PUT and incoming conflicts resolved with local version
* ``SERVER_WINS``: precondition headers and conflicts resolved with remote version

What bothers me is that the sync() should be *recursive* with above policies. Indeed, if we resolve the records implicitly, we should push them back to the server. Doing it twice might not be enough, if two clients are syncing simultaneously for example.

Thoughts ;) ?

File: docs/api.md (hunk 1 above)
Comment (n1k0) [inline docs/api.md]: It's mosty because the line after use `bind`. If you plan on keep this change, it's probably good to update the rest if the file as well :)

**Comment by n1k0** [comment]
>  if we resolve the records implicitly

Just reading this sentence back from 2 weeks away, it's scary as hell :p FWIW, I'd like to keep users handling reconciliation in the most explicit fashion.

**Comment by Natim** [comment]
Actually we should have three kind of conflict handing:

- ``CLIENT_WINS``: no precondition headers in PUT and incoming conflicts resolved with local version
- ``SERVER_WINS``: precondition headers and conflicts resolved with remote version
- ``BOTH_WINS``: Manual conflict handling (we could even try three way merge first)

File: docs/api.md [hunk 2]
```diff
@@ -217,39 +217,53 @@ articles.sync()
...
-  .catch(console.error.bind(console));
+  .then(result => {
+    console.log(result);
+  })
+  .catch(error => {
+    console.error(error);
+  });
 ```
 
+The synchronization updates the local data, and provides information about performed operations.
```
Comment (ametaireau) [inline docs/api.md]: nit: synchronisation.

Comment (ametaireau) [inline docs/api.md]: Actually don't worry, this will be done in a later PR.

**Comment by ametaireau** [comment]
LGTM.
